### DISCUSSION TITLE
A way to search the Variables pane

### DISCUSSION THREAD

**Discussion by jgarte** [inducer/pudb#569] [Ideas]
Is there currently a way to search the variables pane the same way that you can search in the Code pane?

**Comment by inducer** [comment]
There isn't! But I could see that being a useful addition, perhaps as a live-updating filter of sorts. PRs welcome.

**Reply by jgarte** [reply]
@inducer What do you think of having a jump to variables pane for the "variable at point" in the code editor pane? I realize that pudb currently does not have a character by character cursor but only moves by whole lines...

In other words, the feature would be that when you're in the code editor pane if your cursor is under a variable that exists in the variables pane then you can jump to it in the variables pane directly from the code editor pane.

**Reply by jgarte** [reply]
Another idea: a mouse over tooltip pop up for variables in the code editor pane that shows their value

**Reply by inducer** [reply]
I think introducing character-level selection just to point at variables might be a tall order. But the mouse thing could work. I try to avoid using the mouse as much as I can, so it's not likely to be something I'd be interested in, but if someone else were to make it, I likely wouldn't stand in the way.

**Comment by jgarte** [comment]
Another idea is to be able to click on variables in the variables pane in order to fold out the objects and their attributes.
